File: src/mantle/world/CoordTuplePair.scala
package mantle.world

/**
 * CoordTuplePair
 *
 * @param a The first CoordTuple
 * @param b The second CoordTuple
 *
 * @author mDiyo
 * @author Sunstrike <sunstrike@azurenode.net>
 */
class CoordTuplePair(val a:CoordTuple, val b:CoordTuple) {

  def this(aX: Int, aY: Int, aZ: Int, bX: Int, bY: Int, bZ: Int) = this(new CoordTuple(aX, aY, aZ), new CoordTuple(bX, bY, bZ))

}

File: src/mantle/world/CoordTuplePair.java
package mantle.world;

import net.minecraft.util.BlockPos;

/**
 * CoordTuplePair
 *
 * @author mDiyo
 */
public class CoordTuplePair
{
    public BlockPos a;
    public BlockPos b;

    public CoordTuplePair(BlockPos a, BlockPos b)
    {
        this.a = a;
        this.b = b;
    }

    public CoordTuplePair(int aX, int aY, int aZ, int bX, int bY, int bZ)
    {
        this.a = new BlockPos(aX, aY, aZ);
        this.b = new BlockPos(bX, bY, bZ);
    }
}